libs/pika/async_cuda/include/pika/async_cuda/cuda_pool.hpp

```diff
@@ -52,15 +64,24 @@ namespace pika::cuda::experimental {
         locked_cusolver_handle& operator=(locked_cusolver_handle&&) = delete;
         locked_cusolver_handle& operator=(locked_cusolver_handle const&) = delete;
 
+        /// \brief Access the underlying cuSOLVER handle.
+        ///
+        /// \return a reference to the \ref cusolver_handle. The returned handle is not thread-safe
+        /// and must be used within the lifetime of the \ref locked_cusolver_handle.
         PIKA_EXPORT cusolver_handle const& get() noexcept;
     };
 
-    /// A pool of CUDA streams, used for scheduling work on a CUDA device.
+    /// \brief A pool of CUDA streams, used for scheduling work on a CUDA device.
+    ///
+    /// The pool initializes a set of CUDA (thread-local) streams on construction and provides
+    /// access to the streams in a round-robin fashion. The pool also gives access to cuBLAS and
+    /// cuSOLVER handles.
     ///
-    /// The pool initializes a set of CUDA (thread-local) streams on
-    /// construction and provides access to the streams in a round-robin
-    /// fashion. The pool is movable and copyable with reference semantics.
-    /// Copies of a pool still refer to the original pool of streams.
+    /// The pool is movable and copyable with reference semantics.  Copies of a pool still refer to
+    /// the original pool of streams. A moved-from pool can't be used, except to check if it is
+    /// valid with \ref valid().
+    ///
+    /// The pool is equality comparable and formattable.
```

Comment:
Should we keep it simple and document this like above, or should we explicitly add `operator==` etc. to the documentation? I'm leaning towards keep it simple, but curious to hear what others think.

Reply:
Keep it simple. Unless there's special behaviour for special member functions, just mention here that they exist/don't exist.